test/reputation-system/dispute-resolution-misbehaviour.js

```diff
@@ -338,18 +331,230 @@ contract("Reputation Mining - disputes resolution misbehaviour", accounts => {
+        // right taskId, so if they're all created at once it messes up.
+      }
+
+      // We need to complete the current reputation cycle so that all the required log entries are present
+      await advanceMiningCycleNoContest({ colonyNetwork, test: this });
+
+      const clients = await Promise.all(
+        accountsForTest.map(async (addr, index) => {
+          const client = new MaliciousReputationMinerExtraRep(
+            { loader, realProviderPort, useJsTree, minerAddress: addr },
+            accountsForTest.length - index,
+            index
+          );
+          // Each client will get a different reputation update entry wrong by a different amount, apart from the first one which
+          // will submit a correct hash.
+          await client.initialise(colonyNetwork.address);
+          return client;
+        })
+      );
+
+      const repCycle = await getActiveRepCycle(colonyNetwork);
+      await forwardTime(MINING_CYCLE_DURATION / 2, this);
+
+      for (let i = 0; i < 9; i += 1) {
+        // Doing these individually rather than in a big loop because with many instances of the EVM
+        // churning away at once, I *think* it's slower.
+        await clients[i].addLogContentsToReputationTree();
+        await clients[i].submitRootHash();
+        await clients[i].confirmJustificationRootHash();
+      }
+
+      await forwardTime(MINING_CYCLE_DURATION / 2, this);
+
+      console.log("Starting disputes");
+
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[0], clients[1]);
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[2], clients[3]);
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[4], clients[5]);
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[6], clients[7]);
+      await accommodateChallengeAndInvalidateHash(colonyNetwork, this, clients[8]);
+
+      console.log("Starting round 2");
+
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[0], clients[2]);
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[4], clients[6]);
+      await accommodateChallengeAndInvalidateHash(colonyNetwork, this, clients[8]);
+
+      console.log("Cleaning up");
+
+      // Now clean up
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[0], clients[4]);
+      await accommodateChallengeAndInvalidateHash(colonyNetwork, this, clients[8]);
+      await accommodateChallengeAndInvalidateHashViaTimeout(colonyNetwork, this, clients[0], clients[8]);
+      await repCycle.confirmNewHash(4);
+    });
+
+    it("should not mark a round as complete even if a bye was awarded in it", async function advancingTest() {
+      this.timeout(10000000);
+
+      expect(accounts.length, "Not enough accounts for test to run").to.be.at.least(12);
```

Comment:
The next test originally didn't work for me because we didn't have enough accounts; it was nice to have a 'real' error message along those lines but that was the first time I can recall.

Reply:
Hmm I see that. Although was it difficult to figure out what was going on without this error message?

Reply:
You get `     Error: invalid address (arg="guy", coderType="address", value=undefined)` pointing at [this line](https://github.com/JoinColony/colonyNetwork/blob/develop/helpers/test-data-generator.js#L252).

Reply:
Oh that's awful, let's leave this in.